README.md

```diff
@@ -12,10 +12,563 @@ The Chartboost Mediation Reference adapter mediates the Reference SDK via the Ch
 ## Integration
 
 In your `build.gradle`, add the following entry:
-```
+```groovy
     implementation "com.chartboost:chartboost-mediation-adapter-reference:4.1.0.1.0"
 ```
 
+## Chartboost Mediation Custom Adapter Implementation Guidelines
+
+This guide is intended to help you implement custom mediation adapters that work with the Chartboost
+Mediation SDK on Android.
+
+> [!IMPORTANT]
+> Chartboost Mediation will not be providing official support for any custom adapters. For official adapters that we develop and support, [visit this site](https://adapters.chartboost.com).
+
+> [!NOTE]
+> The following instructions are available in Kotlin. If your adapter is going to be published and maintained by Chartboost Mediation, ensure that it is written in Kotlin.
+
+1. Create a new class and implement Chartboost Mediation's `PartnerAdapter` interface.
+
+    ```kotlin
+    class ReferenceAdapter : PartnerAdapter
+    ```
+
+2. Override `val partnerSdkVersion: String` with the version of the partner SDK.
+
+    ```kotlin
+    /**
+     * Override this value to return the version of the partner SDK.
+     */
```

Comment:
We don't need these types of comments if the instructions preceding the code already say the same thing. Here and wherever applicable.